### PR TITLE
[ADF-4227] Material Dialog - change direction option based on textOrientation

### PR DESCRIPTION
**Pull request by pionnegru**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
[ADF-4227](https://issues.alfresco.com/jira/browse/ADF-4227)

**What is the new behaviour?**

previewortl

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/core/services/dialog-config-factory.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+export function dialogConfigFactory(
+    userPreferencesService: UserPreferencesService,
+    dialogConfigService: DialogConfigService
+): any {
+    return () => {
+        return userPreferencesService
+            .select('textOrientation')
+            .subscribe((direction: Direction) => {
+                dialogConfigService.changeDirection(direction);
```
Comment (DenysVuika) [inline lib/core/services/dialog-config-factory.ts]: Can the service listen to those events without this factory?

Comment (pionnegru) [reply lib/core/services/dialog-config-factory.ts]: and how would it be initialized?

Comment (DenysVuika) [reply lib/core/services/dialog-config-factory.ts]: In the service constructor?

Comment (pionnegru) [reply lib/core/services/dialog-config-factory.ts]: I mean, how would the service be initiliezed. the constructor will work if we declare it for example as a dependency of a component. at this stage it won't

Comment (DenysVuika) [reply lib/core/services/dialog-config-factory.ts]: ok, makes sense.

**Pull request merged**